common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAuthority.java

```diff
@@ -173,4 +174,21 @@ public AzureActiveDirectoryAudience getAudience() {
         return mAudience;
     }
 
+    /**
+     * Checks if current authority belongs to the same cloud as the passed in authority
+     *
+     * @param authorityToCheck authority to check against
+     * @return true if host name matches for authorityURL of both authorities, otherwise false
+     */
+    public boolean isSameCloudAsAuthority(@NonNull final AzureActiveDirectoryAuthority authorityToCheck) throws IOException {
```

Comment:
```suggestion
    @WorkerThread
    public boolean isSameCloudAsAuthority(@NonNull final AzureActiveDirectoryAuthority authorityToCheck) throws IOException {
```

`performCloudDiscovery` does network i/o, we should provide some hint you can't call this from any arbitrary thread

Reply:
This annotation requires an import from `androidx.annotation.WorkerThread`

Reply:
I am assuming that CloudDiscovery should have happened by the time we call this method.. But I am not 100% sure if I can make that assumption for all scenarios.